server/api/api/utils.py

```diff
@@ -211,9 +212,12 @@ async def submit_run(
     return response
 
 
-def apply_enrichment_and_validation_on_task(task):
+def apply_enrichment_and_validation_on_task(task, kind):
     # Conceal notification config params from the task object with secrets
     mask_notification_params_on_task(task, server.api.constants.MaskOperations.CONCEAL)
+    task.setdefault("metadata", {}).setdefault("labels", {}).setdefault(
+        mlrun_constants.MLRunInternalLabels.kind, kind
+    )
```

Comment:
Enriching the label here is making the task enrichment even more sparse than it already is. Also this enriches non scheduled jobs that I don't think we want to. We should have a consolidated place that enriches schedules. It should be in `server.api.utils.scheduler.Scheduler` and all of the common enrichments for store/create/update should be there.